smartapps/smartthings/wattvision-manager.src/wattvision-manager.groovy

```diff
@@ -164,9 +174,14 @@ def wattvisionURL(senorId, startDate, endDate) {
 	if (diff > 259200000) { // 3 days in milliseconds
 		// Wattvision only allows pulling 3 hours of data at a time
 		startDate = new Date(hours: endDate.hours - 3)
+	} else if (diff < 10000) { // 10 seconds in milliseconds
+		// Wattvision throws errors when the difference between start_time and end_time is 5 seconds or less
+		// So we are going to make sure that we have a few more seconds of breathing room
+		use (groovy.time.TimeCategory) {
+			startDate = startDate - 10.seconds
```

Comment:
It's a minor point, but if `diff` is 9 sec, then this logic is going to query for a range of 19 sec rather than the minimum of 10. Probably no one will notice but maybe end date minus 10 would be easier to explain.